### PR TITLE
0.6 release

### PR DESCRIPTION
**Pull request by byorgey**
Updates to building instructions, and blog post announcing the release.

### CODE REVIEW

**Comment by byorgey** [comment]
Stuck in an outline of a blog post, but still needs to be filled in with sections highlighting some new features.  I will work on it later, but in the meantime feel free to contribute!

**Comment by byorgey** [comment]
@kostmo would you be willing to write a section or three of the blog post highlighting some of the features you are most proud of in this release?

File: blog/2024-07-14-swarm-0.6-release.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+end
+```
+
+Finally, an embarrassing [bug where variable names sometimes shadowed
+things in outer
+scopes](https://github.com/swarm-game/swarm/issues/681) has been fixed
+via a [massive refactoring that ended up closing a total of *ten*
+issues all at once](https://github.com/swarm-game/swarm/pull/1928).
+
+## WRITE ABOUT MORE FEATURES HERE
```
Comment (kostmo) [inline blog/2024-07-14-swarm-0.6-release.md:59]:
```suggestion
## Cabal sublibraries

A potentially underappreciated feature of Cabal is the ability to split a package into self-contained internal
libraries, known officially as "[sublibraries](https://cabal.readthedocs.io/en/stable/cabal-package-description-file.html#library)".
This has been a powerful tool for us to clean up inter-module dependencies and establish system boundaries. Better disciplined dependencies between modules has sped up incremental compilation and makes it easier to navigate around the code.

We were also able to split off tool functionality that depends on heavyweight external packages (e.g. pandoc) without bogging down compilation of the main game.

[sublibrary dependency diagram goes here]

Refacting into sublibraries starts by selecting a group of modules that is ostensibly self-contained, then iterating with compiler feedback to find where our boundary assumptions are incorrect. This process yields more generic code with loose coupling and high cohesion.

Sometimes it's just a matter of relocating code between modules or introducing a dedicated "Types" module.
In certain cases, we learn that code destined for the sublibrary has references to a "downstream" type; that is, the type is defined in a library that depends on our new sublibrary. We can parameterize the sublibrary code on that type, and just pass it in concretely at the downstream callsite.
For example, we made the code dealing with world composition indenpendent of the `Entity` type (#1836, #1924).

## Structure recognizer

Scenario authors can define "structures"---2D configurations of entities---that can be composed into larger scenes.
New is the ability to recognize when the **player** has assembled one of these predefined structures, which
can be used as a goal criteria or to trigger other events.

Structure recognition is implemented as an efficient 2D [Aho-Corasick](https://en.wikipedia.org/wiki/Aho%E2%80%93Corasick_algorithm) matcher.
Modification of a cell in the world triggers a scan for matches against the library of defined structures.
Metaphorically speaking, a "house" will be recognized exactly when the last "brick" is placed.

```

**Review by kostmo — COMMENTED** [review]
Added a couple of topics: sublibraries and structure recognition.

Am open to refinements for brevity.

File: blog/2024-07-14-swarm-0.6-release.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+![](../gallery/beekeeping.png)
+
+![](../gallery/fishing.png)
+
+![](../gallery/dimsum.png)
+
+## Language improvements
+
+In terms of the Swarm programming language, this release has a few big
```
Comment (nitinprakash96) [inline blog/2024-07-14-swarm-0.6-release.md:40]: I think it's worth to mention that the types in swarm language will now start with uppercase. But I leave that decision upto you.

Comment (byorgey) [reply blog/2024-07-14-swarm-0.6-release.md:40]: Ah, yes, good idea, I will add that.

File: blog/2024-07-14-swarm-0.6-release.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+author: "Brent Yorgey"
+---
+
+The [Swarm](https://github.com/swarm-game/swarm/) development team is
+pleased to announce the latest (alpha) release of the game.
+
+It's been quite a while since our last release, so this one includes a
+lot.  Some of the biggest highlights include native Windows support,
+customizable keybindings, type synonyms and equirecursive types, a ton
+of small UI improvements, and a prototype tournament server where
```
Comment (kostmo) [inline blog/2024-07-14-swarm-0.6-release.md:12]: "small UI improvements" contradicts with "biggest highlights"; can probably remove.  Or re-word without using "small", e.g. "UI polish".

File: blog/2024-07-14-swarm-0.6-release.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+![](../gallery/beekeeping.png)
+
+![](../gallery/fishing.png)
+
+![](../gallery/dimsum.png)
+
+## Language improvements
+
+In terms of the Swarm programming language, this release has a few big
+new features and one breaking change, adds a few more primitive
```
Comment (kostmo) [inline blog/2024-07-14-swarm-0.6-release.md:41]: overuse of "a few"?

File: blog/2024-07-14-swarm-0.6-release.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+commands, and fixes one embarrassing bug.
+
+New primitive commands include `volume` (for measuring enclosed
+areas) and `sow` (for planting things that grow and spread). In
+addition, quite a few primitive commands that were previously only available
```
Comment (kostmo) [inline blog/2024-07-14-swarm-0.6-release.md:46]:
```suggestion
addition, several primitive commands that were previously only available
```

**Review by kostmo — APPROVED** [review]

File: blog/2024-07-14-swarm-0.6-release.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+[release]: https://github.com/swarm-game/swarm/releases
+[hackage]: https://hackage.haskell.org/package/swarm
+[issue]: https://github.com/swarm-game/swarm/issues/new/choose
+
+## Future plans & getting involved
+
+We're still hard at work on the game. Fun upcoming things include:
+
+- More fully fleshed-out [tournament server](https://github.com/swarm-game/swarm/pull/1798)
+- [Saving and loading games][saving]
```
[the remark this answers is not in the document]
Comment (byorgey) [reply blog/2024-07-14-swarm-0.6-release.md:135]: You're not the only one with plans to make progress on it.  I have also started working on it a tiny bit (*e.g.* https://github.com/swarm-game/swarm/issues/1907).  I'd like to leave it there to help goad us into working on it. :smile:

**Review by nitinprakash96 — APPROVED** [review]

**Pull request merged**